x/profile/types/genesis.go

```diff
@@ -50,12 +61,20 @@ func (gs GenesisState) ValidateValidators() error {
 
 		consAddrIndex := string(ValidatorByConsAddressKey(elem.ConsensusAddress))
 		if _, ok := validatorByConsAddressIndexMap[consAddrIndex]; !ok {
-			return fmt.Errorf("validator consensus address not found for ValidatorByConsAddress: %s", elem.Address)
+			return fmt.Errorf("validator consensus address not found for ValidatorByConsAddress: %s", elem.ConsensusAddress)
+		}
+
+		consNonceIndex := string(ConsensusKeyNonceKey(elem.ConsensusAddress))
+		if _, ok := consensusKeyNonceIndexMap[consNonceIndex]; !ok {
+			return fmt.Errorf("validator consensus address not found for ConsensusKeyNonce: %s", elem.ConsensusAddress)
```

Comment:
I think it's ok from the code but just to be sure:

A validator consensus address should always be associated to a validator address and a consensus key nonce should always be associated to a validator consensus address

But a validator address can eventually no be associated with any validator consensus address